rasa/core/policies/policy.py

```diff
@@ -460,6 +460,30 @@ def max_confidence(self) -> float:
         """
         return max(self.probabilities, default=0.0)
 
+    def events_for_prediction(self, action_name: Text) -> List[Event]:
+        """Returns events which should be logged on the tracker for this prediction.
+
+        Args:
+            action_name: The name of the predicted action. If it was an end-to-end
+                prediction the `action_name` is actually the predicted text.
+
+        Returns:
+            The events which should be logged.
+        """
+        action_text = None
+        if self.is_end_to_end_prediction:
+            action_text = action_name
+            action_name = None
```

Comment:
this is incorrect. Whether it is `action_text` or `action_name` decides domain.
`self.is_end_to_end_prediction` only indicates what type of user input (text or intent) was used

Reply:
Yes, of course 😬

Reply:
Too much refactoring - I could luckily just revert the commit 😁